### PR TITLE
UWP CollectionView SelectionMode

### PR DESCRIPTION
**Pull request by humhei**
### Description of Change ###
UWP CollectionView  SelectionMode

### Issues Resolved ### 
#3172
#6489 
#5844
#5625

- fixes #
UWP CollectionView  SelectionMode

### Platforms Affected ### 
- UWP

### API Changes ###

* CollectionViewRender.cs -> (ItemsViewRender.cs + SelectableItemsRender.cs + CollectionViewRender.cs)
~~* Update SelecteableListView.SelectedItems `get`~~
* Changed Render<CollectionView,ItemControl> -> Render<CollectionView,ListViewBase>

image
image
image

### Testing Procedure ###
* CollectionView Gallery
* Enable CollectionView
* Selection Galleries

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by humhei** [comment]
Test most tests in CollectionView gallery,
Seems no back compatible errors occurs.

File: Xamarin.Forms.Core/Items/SelectableItemsView.cs [hunk 1]
```diff
@@ -41,9 +44,10 @@ public object SelectedItem
 			set => SetValue(SelectedItemProperty, value);
 		}
 
+
 		public IList<object> SelectedItems
 		{
-			get => (IList<object>)GetValue(SelectedItemsProperty);
+			get => _selectedItems;
```
Comment (hartez) [inline Xamarin.Forms.Core/Items/SelectableItemsView.cs]: This breaks binding for `SelectedItems`. If you want to see the break in action, open Control Gallery on Android or iOS, navigate to Collection View Gallery -> Selection Galleries -> Multiple Selection, Bound and tap the "Clear CV selection and add items 0 and 3" button. In the list, only items 0 and 3 should be selected. With these changes to SelectableItemsView, no items are selected.

Comment (hartez) [reply Xamarin.Forms.Core/Items/SelectableItemsView.cs]: AFAICT, none of the changes in SelectableItemsView here are necessary to implement selection on UWP.

**Comment by humhei** [comment]
> In the pre-selected items test, the initial items are selected. But if I tap any other item, the previously selected items are cleared.
> In the selection modes test, if I select Multiple, I still can only select single items.

This is because that the `Xamarin.Core Multiple` is corresponding to `UWP` `Extend` mode

In UWP `extend` mode,
You have to press the CTRL Key while select to other items



image
Preselected Items test, with CTRL Pressed

**Comment by humhei** [comment]
OK,
I will change UWP `Extended` to `Multiple` mode.

**Comment by humhei** [comment]
> This breaks binding for SelectedItems. If you want to see the break in action, open Control Gallery on Android or iOS, navigate to Collection View Gallery -> Selection Galleries -> Multiple Selection, Bound and tap the "Clear CV selection and add items 0 and 3" button. In the list, only items 0 and 3 should be selected. With these changes to SelectableItemsView, no items are selected.

This issue  exists before this PR.

This is because the initial empty is a List
https://github.com/xamarin/Xamarin.Forms/blob/4a756b9b982b5a785d9d4b3d5fc9bb63b3cf00b9/Xamarin.Forms.Core/Items/SelectableItemsView.cs#L32

Which is not SelectionList()
image

I will try to fix it

**Comment by humhei** [comment]
@hartez  Should get worked now.

**Pull request merged**